lib/ace/mouse/touch_handler.js

```diff
@@ -47,12 +48,99 @@ exports.addTouchListeners = function(el, editor) {
     var vX = 0;
     var vY = 0;
     var pressed;
+    var contextMenu;
+    
+    function showContextMenu() {
+        var clipboard = window.navigator && window.navigator.clipboard;
+        var isOpen = false;
+        if (!contextMenu) {
+            var updateMenu = function() {
```

Comment:
Consider putting the code in this `if` in a method, like: `if (!contextMenu) { contextMenu = createContextMenu(..) }`